diff --git a/slsDetectorGui/CMakeLists.txt b/slsDetectorGui/CMakeLists.txt
index 55c68ca6..2c6efe55 100644
--- a/slsDetectorGui/CMakeLists.txt
+++ b/slsDetectorGui/CMakeLists.txt
@@ -112,7 +112,8 @@ set_target_properties(zmq PROPERTIES
 )
 
 target_link_libraries(slsDetectorGui 
-    slsDetectorShared
+    SlsDetector
+    SlsReceiver
     ${QT_QTCORE_LIBRARIES}
     ${QT_QTGUI_LIBRARIES}
     ${QWT_LIBRARIES}
