diff --git a/slsDetectorGui/CMakeLists.txt b/slsDetectorGui/CMakeLists.txt
index bad299e1..3c980dfb 100644
--- a/slsDetectorGui/CMakeLists.txt
+++ b/slsDetectorGui/CMakeLists.txt
@@ -107,7 +107,8 @@ set_target_properties(slsDetectorGui PROPERTIES
 
 
 target_link_libraries(slsDetectorGui 
-    slsDetectorShared
+    SlsDetector
+    SlsReceiver
     ${QT_QTCORE_LIBRARIES}
     ${QT_QTGUI_LIBRARIES}
     ${QWT_LIBRARIES}
